src/adsp_config.c

```diff
@@ -2337,18 +2338,69 @@ static int adsp_parse_config_fd(FILE *fd, struct image *image)
 	return ret;
 }
 
+static FILE* adsp_conf_files_open(struct adsp_conf_files *files, const char *mode)
+{
+	struct stat statbuf;
+	char *buf;
+	int ret;
+	int i;
+	size_t file_size[MAX_SUPPORTED_CONF_FILES];
+	size_t buf_size = 0;
+
+	for (i = 0; i < files->file_count; i++) {
+		ret = stat(files->file[i], &statbuf);
+		if (ret) {
+			fprintf(stderr, "error: failed to stat file %s\n", files->file[i]);
+			return NULL;
+		}
+
+		file_size[i] = statbuf.st_size;
+		buf_size += statbuf.st_size;
+	}
+	/* Reserve space for adding '\n' at the end of each file */
+	buf_size += files->file_count;
+	files->buffer = malloc(buf_size);
+
+	buf = files->buffer;
+	for (i = 0; i < files->file_count; i++) {
+		FILE *file;
+		size_t read;
+
+		file = fopen(files->file[i], mode);
+		if (!file) {
+			fprintf(stderr, "error: failed to open file %s\n", files->file[i]);
+			return NULL;
+		}
+		read = fread(buf, 1, file_size[i], file);
+		if (read != file_size[i]) {
+			fprintf(stderr, "error: failed to read file %s\n", files->file[i]);
+			return NULL;
+		}
+		fclose(file);
+
+		buf[file_size[i]] = '\n';
+		buf += file_size[i] + 1;
+	}
+	return fmemopen(files->buffer, buf_size, mode);
```

Comment:
Ok this is quite clever approach. "fmemopen()" will potentially hit some compatibility issues, so needs a smoketest on Windows at minimum.

I was actually thinking of something much more low-tech. I.e. "cat a.toml b.toml >$BUILD/rimage.toml" in make rules, and pass generated "rimage.toml" to rimage. But making this in a portable way is not zero effort either, so I guess depends in the end what is the lowest effort way to make this so that it works across platforms.

Reply:
A new proposal, see #179, this probably won't work on windows, let's close